### PR TITLE
Update ADC documentation

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: docs/api/IoT.js-API-ADC.md [hunk 1]
```diff
@@ -10,47 +10,32 @@ The following shows ADC module APIs available for each platform.
...
-* `configuration <Object>`
-  * `device <String>`, mandatory configuration on Linux
-  * `pin <Number>`, mandatory configuration on Nuttx
-* `callback <Function(err: Error | null)>`
-* Returns: `<ADCPin>`
+### adc.open(configuration[, callback])
+* `configuration` {Object}
+  * `device` {String} mandatory configuration on Linux
+  * `pin` {Int} mandatory configuration on Nuttx
+* `callback` {Function(err: Error | null)}
```
Comment (robertsipka) [inline docs/api/IoT.js-API-ADC.md]: The correct style is:
* `callback` {Function}
  * `err` {Error|null} 
Please fix all instances.

File: docs/api/IoT.js-API-ADC.md [hunk 2]
```diff
@@ -10,47 +10,32 @@ The following shows ADC module APIs available for each platform.
...
 
-### `adc.open(configuration[, callback])` <a name="adc-open"></a>
-* `configuration <Object>`
-  * `device <String>`, mandatory configuration on Linux
-  * `pin <Number>`, mandatory configuration on Nuttx
-* `callback <Function(err: Error | null)>`
-* Returns: `<ADCPin>`
+### adc.open(configuration[, callback])
+* `configuration` {Object}
+  * `device` {String} mandatory configuration on Linux
```
Comment (robertsipka) [inline docs/api/IoT.js-API-ADC.md]: {string} Mandatory ...

File: docs/api/IoT.js-API-ADC.md [hunk 3]
```diff
@@ -10,47 +10,32 @@ The following shows ADC module APIs available for each platform.
...
-    * [`adcpin.closeSync()`](#adcpin-close-sync)
-
-## `pin` <a name="pin"></a>
-On Nuttx, you have to know pin number that is defined in target board module. For more module information, please see below list.
+### `pin`
+On Nuttx, you have to know the number of pins that is defined on the target board module. For more information, please see the list below.
   * [STM32F4-discovery](../targets/nuttx/stm32f4dis/IoT.js-API-Stm32f4dis.md#adc-pin)
 
-## Class: ADC <a name="adc"></a>
+# Class: ADC
```
Comment (zherczeg) [inline docs/api/IoT.js-API-ADC.md]: Would be good to add a few sentences what exactly this class do.

File: docs/api/IoT.js-API-ADC.md [hunk 4]
```diff
@@ -10,47 +10,32 @@ The following shows ADC module APIs available for each platform.
...
 
 
-## Constructor <a name="adc-constructor"></a>
+## Constructor
```
Comment (zherczeg) [inline docs/api/IoT.js-API-ADC.md]: Delete constructor section header.

File: docs/api/IoT.js-API-ADC.md [hunk 5]
```diff
@@ -10,47 +10,32 @@ The following shows ADC module APIs available for each platform.
...
-### `adc.open(configuration[, callback])` <a name="adc-open"></a>
-* `configuration <Object>`
-  * `device <String>`, mandatory configuration on Linux
-  * `pin <Number>`, mandatory configuration on Nuttx
-* `callback <Function(err: Error | null)>`
-* Returns: `<ADCPin>`
+### adc.open(configuration[, callback])
+* `configuration` {Object}
+  * `device` {String} mandatory configuration on Linux
+  * `pin` {Int} mandatory configuration on Nuttx
```
Comment (robertsipka) [inline docs/api/IoT.js-API-ADC.md]: {int} Mandatory  ...

**Comment by daeyeon** [comment]
Please check `Common Checklists` in #873 for updating.

File: docs/api/IoT.js-API-ADC.md [hunk 6]
```diff
@@ -10,47 +10,28 @@ The following shows ADC module APIs available for each platform.
...
-  * [Prototype methods](#adcpin-prototype-methods)
-    * [`adcpin.read([callback])`](#adcpin-read)
-    * [`adcpin.readSync()`](#adcpin-read-sync)
-    * [`adcpin.close([callback])`](#adcpin-close)
-    * [`adcpin.closeSync()`](#adcpin-close-sync)
-
-## `pin` <a name="pin"></a>
-On Nuttx, you have to know pin number that is defined in target board module. For more module information, please see below list.
+### `pin`
+On Nuttx, you have to know the number of pins that is defined on the target board module. For more information, please see the list below.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-ADC.md]: I would put this text under the `## Class: ADC`, remove the `### pin` header, and add a bit more text about what this component does.

**Comment by DanielBallaSZTE** [comment]
@zherczeg I've updated the document.

**Comment by daeyeon** [comment]
LGTM

**Pull request merged**